hkmc2/shared/src/main/scala/hkmc2/syntax/Tree.scala

```diff
@@ -202,6 +202,12 @@ enum Tree extends AutoLocated:
     case Spread(Keyword.`...`, _, S(und: Under)) => S(S(true), new Ident("_").withLocOf(und), N)
     case InfixApp(lhs: Ident, Keyword.`:`, rhs) => S(N, lhs, S(rhs))
     case TermDef(ImmutVal, inner, _) => inner.asParam
+    case Modified(Keyword.`using`, _, inner) => inner match
+      // Param of form (using ..., name: Type). Parse it as usual.
+      case inner: InfixApp => inner.asParam
+      // Param of form (using ..., Type). Synthesize an identifier for it.
+      // TODO: Synthesize a better name.
+      case _ => S(N, Ident(inner.showDbg.replaceAll("\\W", "_")), S(inner))
```

Comment:
That's _cursed_. Why do you need a weird name for this? Just use something like
```suggestion
      case _ => S(N, Ident("$ev")), S(inner))
```
Or maybe just an empty string, which could be useful later on to check whether there was an actual name given.
The name doesn't need to be unique.

Reply:
A long time ago when I created the PR, this didn't work because JS complained about the identifier containing illegal characters. Now it seems to work fine. I'll remove the `replaceAll` part.

But I don't understand why doesn't the name need to be unique. If the name is not unique, we won't be able to define multiple `use`s in one block, e.g.,

```fs
use Bar[Int, Str] = new IntStrBar()
use Bar[Str, Int] = new StrIntBar()
//│ FAILURE: Unexpected type error
//│ ╔══[ERROR] Multiple definitions of symbol 'instance'
//│ ╟── defined here
//│ ║  l.99: 	use Bar[Int, Str] = new IntStrBar()
//│ ║        	    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
//│ ╟── defined here
//│ ║  l.100: 	use Bar[Str, Int] = new StrIntBar()
//│ ╙──       	    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

which might be a common use case.